lib/matplotlib/cm.py

```diff
@@ -193,6 +187,39 @@ def unregister(self, name):
                              "colormap.")
         self._cmaps.pop(name, None)
 
+    def get_cmap(self, cmap):
+        """
+        Ensure that at given object is a converted to a color map.
+
+        If *cmap* in `None`, returns the Colormap named by :rc:`image.cmap`.
+
+        Parameters
+        ----------
+        cmap : str, Colormap, None
+
+            - if a `~matplotlib.colors.Colormap`, return it
+            - if a string, look it up in mpl.colormaps
+            - if None, look up the default color map in mpl.colormaps
+
+        Returns
+        -------
+        Colormap
+
+        Raises
+        ------
+        KeyError
```

Comment:
We may want to transform this into a ValueError. `get_cmap` is less a lookup and more a transformation function.

Reply:
I have a slight preference for `KeyError` to mirror `__getitem__`.  Maybe it should raise both and add another check that `cmap` is actually a string and then raise a `ValueError`?  There are ways to call this now that will get "not hashable" errors etc.

Reply:
If you're going to check strings, then I'd think it would be `if isinstance(cmap, str): return valid or raise ValueError; else: raise TypeError`, and maybe no `KeyError` at all?